tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml

```
@@ -0,0 +1,930 @@
+  #    state: present
+  #    vpc_id: "{{ vpc_id }}"
+  #    inherit_existing: False
+  #    delete_old: True
+  #    ntp_servers:
+  #        - 10.0.0.2
+  #        - 10.0.1.2
+  #    netbios_name_servers:
+  #        - 10.0.0.1
+  #        - 10.0.1.1
+  #    tags:
+  #      AnsibleTest: integration
+  #      Collection: amazon.aws
+  #  register: dhcp_options
+  #  check_mode: true
+
+  #- assert:
+  #    that:
+  #      - dhcp_options.changed
+
+  - name: add and remove tags
+    ec2_vpc_dhcp_option:
+      dhcp_options_id: "{{ dhcp_options.dhcp_options_id }}"
+      state: present
+      vpc_id: "{{ vpc_id }}"
+      inherit_existing: False
+      delete_old: True
+      ntp_servers:
+          - 10.0.0.2
+          - 10.0.1.2
+      netbios_name_servers:
+          - 10.0.0.1
+          - 10.0.1.1
+      tags:
+        AnsibleTest: integration
+        Collection: amazon.aws
+    register: dhcp_options
+
+  - name: check if the expected tags are associated
+    ec2_vpc_dhcp_option_info:
+      dhcp_options_ids: ["{{ dhcp_options.dhcp_options_id }}"]
+    register: dhcp_options_info
+
+  - assert:
+      that:
+        - dhcp_options.changed
+        - not dhcp_options.new_options
+        - new_dhcp_options_id == dhcp_options.dhcp_options_id
+        - dhcp_options_info.dhcp_options[0].tags is defined
+        - dhcp_options_info.dhcp_options[0].tags.keys() | length == 2
+        - dhcp_options_info.dhcp_options[0].tags['collection'] == 'amazon.aws'
+        - dhcp_options_info.dhcp_options[0].tags['ansible_test'] == 'integration'
+
+  - name: add tags with different cases
+    ec2_vpc_dhcp_option:
+      dhcp_options_id: "{{ dhcp_options.dhcp_options_id }}"
+      state: present
+      vpc_id: "{{ vpc_id }}"
+      inherit_existing: False
+      delete_old: True
+      ntp_servers:
+          - 10.0.0.2
+          - 10.0.1.2
+      netbios_name_servers:
+          - 10.0.0.1
+          - 10.0.1.1
+      tags:
+        "lowercase spaced": 'hello cruel world'
+        "Title Case": 'Hello Cruel World'
+        CamelCase: 'SimpleCamelCase'
+        snake_case: 'simple_snake_case'
+    register: dhcp_options
+
+  - name: check if the expected tags are associated
+    ec2_vpc_dhcp_option_info:
+      dhcp_options_ids: ["{{ dhcp_options.dhcp_options_id }}"]
+    register: dhcp_options_info
+
+  - assert:
+      that:
+        - dhcp_options.changed
+        - not dhcp_options.new_options
+        - new_dhcp_options_id == dhcp_options.dhcp_options_id
+        - dhcp_options_info.dhcp_options[0].tags is defined
+        - dhcp_options_info.dhcp_options[0].tags.keys() | length == 4
+        - dhcp_options_info.dhcp_options[0].tags['lowercase spaced'] == 'hello cruel world'
+        - dhcp_options_info.dhcp_options[0].tags['title _case'] == 'Hello Cruel World'
+        - dhcp_options_info.dhcp_options[0].tags['camel_case'] == 'SimpleCamelCase'
+        - dhcp_options_info.dhcp_options[0].tags['snake_case'] == 'simple_snake_case'
```

The middle 2 are bugs: it should be coming back as 
```suggestion
        - dhcp_options_info.dhcp_options[0].tags['Title Case'] == 'Hello Cruel World'
        - dhcp_options_info.dhcp_options[0].tags['CamelCase'] == 'SimpleCamelCase'
        - dhcp_options_info.dhcp_options[0].tags['snake_case'] == 'simple_snake_case'
```

I guess we comment them out and start working on a v3 implementation with a pile of bug fixes...